### PR TITLE
feat: button component with storybook docs

### PR DESCRIPTION
**Pull request by supminn**
# Description
- Button component has 3 types:
  -  Pill button - Button that contains text value. It is commonly used for call to action, tags or filters.
  - Pill button with icons - Button containing icon either on the left or right side of the button. 
  - Icon button - Button is represented by just an icon without any text. 
- They have 3 variants - Primary, Secondary and Tertiary. 
- Different states a button can have are as follows: Active, Disabled, SoftDisabled (The button is currently disabled for user interaction; however, it remains focusable.)
- Colors:
  - Positive - Green button 
  - Negative - Red button 
  - Accent - Blue button 
  - Promotional - Purple button 
  - Default - Grey button 

We support the following sizes which vary based on the button type. As per our conventions guide, we shall use REM unit for button sizing.
- Pill button - 40, 32, 28, 24. 
- Icon button - 64, 52, 40, 32, 28 and 24.
    - Tertiary icon button can be of 20.
 
## Links
- [Tech Breakdown](https://confluence-eng-gpk2.cisco.com/conf/pages/viewpage.action?pageId=569871745)
- [Component and styling ticket](https://jira-eng-gpk2.cisco.com/jira/browse/MOMENTUM-386)
- [Storybook ticket](https://jira-eng-gpk2.cisco.com/jira/browse/MOMENTUM-387)

### Pill Button
image

### Pill with Icon Button (prefix slot)
image

### Pill with Icon Button (postfix slot)
image

### Icon Button (soft disabled)
image

### Active disabled button
image

### CODE REVIEW

File: packages/components/config/plop/templates/add/component/{{componentName}}.component.ts.hbs [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { Component } from '../../models';
 /**
  * {{componentName}} component, which ...
  *
- * @tagname {{componentName}}
+ * @tagname {{prefix}}{{separator}}{{componentName}}
```
Comment (supminn) [inline packages/components/config/plop/templates/add/component/{{componentName}}.component.ts.hbs:8]: Added a small change to the plop script template.

File: packages/components/config/plop/templates/add/component/{{componentName}}.stories.ts.hbs [hunk 2]
```diff
@@ -8,6 +8,7 @@ const render = (args: Args) => html`
 
 const meta: Meta = {
   title: 'Work In Progress/{{componentName}}',
+  tags: ['autodocs'],
```
Comment (supminn) [inline packages/components/config/plop/templates/add/component/{{componentName}}.stories.ts.hbs:11]: Added a small change to the plop script template.

File: packages/components/src/components/button/button.e2e-test.ts
Comment (supminn) [inline packages/components/src/components/button/button.e2e-test.ts:1]: This is just a template file, no changes have been made to this file.

File: packages/components/src/components/button/button.component.ts [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+
+  /**
+   * Handles the keydown event on the button.
+   * If the key is 'Enter' or 'Space', the button is pressed.
+   *
+   * @param event - The keyboard event.
+   */
+  private handleKeyDown(event: KeyboardEvent) {
+    if (!this.disabled && (event.key === 'Enter' || event.key === ' ')) {
+      event.preventDefault();
```
Comment (NotNestor) [inline packages/components/src/components/button/button.component.ts:330]: I think we should add a utility for if it is an action key. This check is also missing the NumpadEnter key which should also activate the button.

I would also suggest adding constants for the keycodes we need to check

File: packages/components/src/components/button/button.component.ts [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+ * - **Primary**: Solid background color.
+ * - **Secondary**: Transparent background with a solid border.
+ * - **Tertiary**: No background or border, appears as plain text but retains all button functionalities.
+ *
+ * Button Colors:
+ * - **Positive**: Green color.
+ * - **Negative**: Red color.
+ * - **Accent**: Blue color.
+ * - **Promotional**: Yellow color.
+ * - **Active**: White color.
```
Comment (NotNestor) [inline packages/components/src/components/button/button.component.ts:26]: Promotional is not a yellow colour it is Magenta

File: packages/components/src/components/button/button.component.ts [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+ * @dependency mdc-icon
+ *
+ * @tagname mdc-button
+ *
+ * @slot prefix-icon - Icon positioned to the left of the button text or within an icon button.
+ * @slot default slot - Text slot of the button.
+ * @slot postfix-icon - Icon positioned to the right of the button text.
+ */
+class Button extends Component {
+  /**
```
Comment (NotNestor) [inline packages/components/src/components/button/button.component.ts:51]: Does it make sense to have a base button class that does some of this logic but is not styled exactly likes. For example there is a proposal to implement tabs in the contact centre as rounded buttons
Screenshot 2024-11-08 at 17 03 02

**Comment by supminn** [comment]
Closing this PR and re-opening separate PRs for component and storybook.

**Pull request closed without merging**